### ISSUE TITLE
Unknown Layer Error - While running demo.py

### ISSUE THREAD

**Issue by karthee320** [tenstep/DeepModel#13]
Hi,

I have copied your files from ./libs/include to caffe_root/include and ./libs/src to caffe_root/src and built the caffe library(make && make pycaffe) by enabling the python layer WITH_PYTHON_LAYER := 1 in Makefile.config. Even then it is reporting an error stating no layer named DeepHandModel. Can you please help in resolving this issue.

F0606 19:27:40.747319 20872 layer_factory.hpp:80] Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: DeepHandModel (known types: AbsVal, Accuracy, ArgMax, BNLL, BatchNorm, Concat, ContrastiveLoss, Convolution, Data, Deconvolution, Dropout, DummyData, Eltwise, Embed, EuclideanLoss, Exp, Filter, Flatten, HDF5Data, HDF5Output, HingeLoss, Im2col, ImageData, InfogainLoss, InnerProduct, LRN, Log, MVN, MemoryData, MultinomialLogisticLoss, PReLU, Pooling, Power, Python, ReLU, Reduction, Reshape, SPP, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, Softmax, SoftmaxWithLoss, Split, TanH, Threshold, Tile, WindowData)
*** Check failure stack trace: ***
Aborted (core dumped)

**Comment by xingyizhou** [comment]
Hi Karthee320, 
  Thanks for your interests on our project. The code should be able to run correctly if you strictly follow the instructions. Please check the following things: 
- Make sure you have copied the entire folder `libs/include/caffe/*.h` instead of individual files `libs/include/*.h`
- Make sure you have set the corresponding `pycaffe_root` in `path.config`. Note that in Linux  the default `pycaffe_root` is `caffe_root/python/`

**Comment by sainisanjay** [comment]
Hi @karthee320 same error i am getting any idea how do you solved. @xingyizhou i have copied all files properly but than i am getting same error.  The error stating  Unknown layer type: DeepHandModel. However, i have checked `deep_hand_model_layers.cpp ` file is already available in `..DeepModel-master/caffe/src/caffe`

**Comment by sainisanjay** [comment]
Dear @xingyizhou as you have added new layer `deep_hand_model_layers` thus i guess first we have to register this layer into the Caffe (`caffe.proto`) than only caffe will recognize this layer. I guess thts why error is coming `Unknown layer type: DeepHandModel`

**Comment by sainisanjay** [comment]
Dear @xingyizhou  as you have added new layer `deep_hand_model_layers` thus i guess first we have to register this layer into the Caffe (`caffe.proto`) than only caffe will recognize this layer. I guess thts why error is coming `Unknown layer type: DeepHandModel`
